sections/Chapter 5 Sprint 3/Validation.tex

```diff
@@ -0,0 +1,81 @@
+\section{Validation}
+
+As mentioned in section \ref{databaseResponsibility}, the knowledge layer posts data to the database that later need to be fetched by the functionality layer.
+
+To ensure that the data being posted is correct, we have developed a validation method in collaboration with the knowledge layer.
+To validate the JSON files received from the knowledge layer, we created a JSON schema that specifies the structure that the JSON files can be checked against. 
+Code snippet \ref{lst:jsonSchemaExample} shows an example of a JSON schema that validates JSON objects modelling a simple product in a store. In this example, it is worth noting that the properties \texttt{name} and \texttt{price} are required. Any JSON objects that do not have these properties are invalid when checked against this schema. The \texttt{\$schema} field specifies the standard that the schema is written according to; in this case, the draft v4 specification \cite{tutorialspoint_jsonschema}.
+
+\begin{lstlisting}[language=CSharp, caption={An example of a basic JSON schema.}, label={lst:jsonSchemaExample}]
+{
+   "$schema": "http://json-schema.org/draft-04/schema#",
+   "title": "Product",
+   "type": "object",
+	
+   "properties": {		
+      "name": {
+         "description": "Name of the product",
+      },
+		
+      "price": {
+         "type": "number",
+         "minimum": 0,
+      }
+   },
+	
+   "required": ["name", "price"]
+}
+\end{lstlisting}
+
+We first created a new controller called \texttt{SchemaController} which is responsible for posting JSON schemas to the database. 
+Currently, only one schema is stored in the database, however the system is developed such that it is scalable and expandable.
```

Comment:
Include a little addition as to how it is scalable.
Could be something like this. 
```suggestion
Currently, only one schema is stored in the database, however the system is developed to be scalable as more schemas could be added in the database and validation for these could be created in the \texttt{SchemaController}.
```